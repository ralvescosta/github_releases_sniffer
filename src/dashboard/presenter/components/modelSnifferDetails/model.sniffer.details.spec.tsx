import React from 'react';
import {render} from '@testing-library/react-native';

import {ModelSnifferDetails} from './index';
import {SniffedGithubRepositoryEntity} from '../../../bussiness/entities/sniffed.github.repository.entity';
import {ModalSnifferDetailsViewControllerSpy} from '../../__test__/modal.sniffer.details.view.controller.spy';

function makeSut() {
  const WrapperModal = () => {
    const repository = new SniffedGithubRepositoryEntity(1, '', '', 10, 10, 10, 1, '', '', '');
    const modalControllerSpy = new ModalSnifferDetailsViewControllerSpy();
    return <ModelSnifferDetails repository={repository} viewController={modalControllerSpy} />;
  };

  return render(<WrapperModal />);
}

describe('Model Sniffer Details Component', () => {
  it('Should render correctly', () => {
    const sut = makeSut();

    expect(sut.toJSON()).toMatchSnapshot();
  });
});
