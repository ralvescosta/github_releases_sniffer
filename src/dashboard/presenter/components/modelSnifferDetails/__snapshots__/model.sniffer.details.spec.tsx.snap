// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Model Sniffer Details Component Should render correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  visible={true}
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "padding": 22.5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 34,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ÔÄç
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "space-around",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "width": "100%",
          }
        }
      >
        <Image
          resizeMode="cover"
          source={
            Object {
              "uri": "",
            }
          }
          style={
            Object {
              "height": 85,
              "width": 85,
            }
          }
        />
        <Text
          style={
            Object {
              "fontSize": 60,
              "fontWeight": "bold",
              "marginTop": 15,
            }
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 36,
            }
          }
        >
           Latest release: 
        </Text>
        <Text
          style={
            Object {
              "fontSize": 36,
              "fontWeight": "bold",
            }
          }
        >
          
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "elevation": 5,
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text>
          Remove Repository on Sniffer
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
